activesupport/lib/active_support/core_ext/module/delegation.rb

```diff
@@ -199,7 +199,35 @@ def delegate(*methods, to: nil, prefix: nil, allow_nil: nil, private: nil)
 
       # Attribute writer methods only accept one argument. Makes sure []=
       # methods still accept two arguments.
-      definition = /[^\]]=\z/.match?(method) ? "arg" : "..."
+      definition = \
+        if /[^\]]=\z/.match?(method)
+          "arg"
+        else
+          method_object =
+            begin
+              if to.is_a?(Module)
+                to.method(method)
+              elsif receiver == "self.class"
+                method(method)
+              end
+            rescue NameError
+              # Do nothing. Fall back to `"..."`
+            end
+
+          if method_object
+            parameters = method_object.parameters
+
+            if (parameters.map(&:first) & [:opt, :rest, :keyreq, :key, :keyrest]).any?
```

Comment:
```suggestion
            if parameters.any? { |type, _| [:opt, :rest, :keyreq, :key, :keyrest].include?(type) }
```

Boot performance isn't quite as important as runtime perf, but this seem a bit wasteful to me.

Reply:
Interestingly, this seems untrue, because the Array literal in the block allocates a new Array per each iteration.

```ruby
arr = [*1..10]

count = ObjectSpace.count_objects[:T_ARRAY]
(arr & [10, 11, 12]).any?
p new_arr1: ObjectSpace.count_objects[:T_ARRAY] - count

count = ObjectSpace.count_objects[:T_ARRAY]
arr.any? { |i| [10, 11, 12].include?(i) }
p new_arr2: ObjectSpace.count_objects[:T_ARRAY] - count
```
```
{:new_arr1=>2}
{:new_arr2=>10}
```

Perhaps the most efficient code is to define `[:opt, :rest, :keyreq, :key, :keyrest]` as an ivar outside of the block, but I think that's too much. Maybe it's time to propose a new feature `# frozen_array_literal: true`? 😝

Reply:
Arf, I can never remember which array methods have been optimized to avoid allocations...

> Maybe it's time to propose a new feature # frozen_array_literal: true?

I think it already exist: https://ruby-doc.org/core-3.0.2/doc/syntax/comments_rdoc.html#label-shareable_constant_value+Directive